unit_test/test_acknowledgements.py

```diff
@@ -0,0 +1,275 @@
+
+    def test_four_volunteer_names(self):
+        '''
+        Test case for the situation where there are four volunteer names.
+        '''
+
+        with TemporaryDirectory() as temp_folder:
+            data_folder = Path(os.getcwd(), "unit_test", "data")
+            self.assertTrue(
+                data_folder.is_dir(),
+                "Subdirectory data of directory unit_test is missing.")
+            csv_file_path = Path(data_folder, "volunteer_acknowledgements.csv")
+            output_file_path = Path(temp_folder, "final_step.md")
+            output_file_path.touch()
+            language = "nl-NL"
+            volunteers = ["Volunteer One", "Vrijwilliger Twee",
+                          "Freiwillige Drei", "Voluntario Cuatro"]
+            logging = "off"
+            result = nttt.acknowledgements.add_volunteer_acknowledgement(
+                csv_file_path, output_file_path, language, volunteers, logging)
+            self.assertTrue(result)
+
+            # Check contents of final_step.md.
+            contents = output_file_path.read_text()
+            lines = contents.splitlines()
+            self.assertEqual(13, len(lines))
+            self.assertEqual("", lines[0])
+            self.assertEqual("***", lines[1])
+            self.assertEqual(self.FIRST_LINE, lines[2])
+            self.assertEqual("", lines[3])
+            self.assertEqual("Volunteer One", lines[4])
+            self.assertEqual("", lines[5])
+            self.assertEqual("Vrijwilliger Twee", lines[6])
+            self.assertEqual("", lines[7])
+            self.assertEqual("Freiwillige Drei", lines[8])
+            self.assertEqual("", lines[9])
+            self.assertEqual("Voluntario Cuatro", lines[10])
+            self.assertEqual("", lines[11])
+            self.assertEqual(self.LAST_LINE, lines[12])
+
+    def test_contents_preserved(self):
+        '''
+        Test case for the situation where the output file has contents that
+        have to be preserved.
+        '''
+
+        with TemporaryDirectory() as temp_folder:
+            data_folder = Path(os.getcwd(), "unit_test", "data")
+            self.assertTrue(
+                data_folder.is_dir(),
+                "Subdirectory data of directory unit_test is missing.")
+            csv_file_path = Path(data_folder, "volunteer_acknowledgements.csv")
+            output_file_path = Path(temp_folder, "final_step.md")
+            output_file_path.write_text("Line One.\nLine 2,\nline three.\n")
+            language = "nl-NL"
+            volunteers = ["Volunteer One"]
+            logging = "off"
+            result = nttt.acknowledgements.add_volunteer_acknowledgement(
+                csv_file_path, output_file_path, language, volunteers, logging)
+            self.assertIsNotNone(result)
+
+            # Check contents of final_step.md.
+            contents = output_file_path.read_text()
+            lines = contents.splitlines()
+            self.assertEqual(9, len(lines))
```

Comment:
Same here. Comparing to a string would be more readable.

Reply:
Agreed; simplified